bot/exts/utilities/bookmark.py

```diff
@@ -117,6 +137,40 @@ async def action_bookmark(
         else:
             log.info(f"{member} bookmarked {target_message.jump_url} with title '{title}'")
 
+    @staticmethod
+    async def user_is_permitted_to_bookmark(
```

Comment:
I'm not sure this function had to be extracted out into it's own function. It also does more than the name implies, not only does it *check* if a user is permitted to bookmark, but also sends an error message.

I think we should just move the logic back into the context menu callback and original text command, since a method to build the error embed is already there:
```py
if not permissions.read_messages:
    log.info(...)
    error_embed = Bookmark.build_error_embed(...)

    # In the context menu callback...
    interaction.response.send_message(embed=error_embed, ephemeral=True)

    # OR in the text-based command:
    await ctx.send(embed=error_embed)
```
The way we have it now, the interaction would timeout and error because we use `ctx.send` rather than replying to the interaction with `interaction.reply.send_message`